test/e2e/aaas.go

```diff
@@ -0,0 +1,158 @@
+		Expect(resp.StatusCode).To(Equal(http.StatusOK))
+		s, err := getBody(resp)
+		framework.ExpectNoError(err)
+		Expect(s).To(Equal("Got response"))
+	})
+
+	It("opaServeResponse filter should return body [Ingress] [Opa]", func() {
+		// Authorization done with the rule
+		// https://github.bus.zalan.do/corporate-iam/aaas-infrastructure-tests-policies/blob/main/bundle/policy/ingress/permissions.rego
+
+		serveResponsePath := "/permissions"
+		expectedPermissionsContent := "\"permissions\":{\"permission1\":{},\"permission2\":{}}"
+		path := serveResponsePath
+		ingressRoute := fmt.Sprintf(`authorize: PathRegexp("%s") -> opaServeResponse("%s") -> <shunt>;`, serveResponsePath, opaPolicyName)
+		ingressUpdate := updateIngressAndWait(serviceName, hostName, path, ingressRoute, port, ingressCreate, cs)
+
+		By(fmt.Sprintf("Calling ingress %s/%s we wait to get a 403 with opaServeResponse %s policy", ingressUpdate.Namespace, ingressUpdate.Name, opaPolicyName))
+		rt, quit := createHTTPRoundTripper()
+		defer func() {
+			quit <- struct{}{}
+		}()
+
+		url := "https://" + hostName + serveResponsePath
+		req, err := http.NewRequest("GET", url, nil)
+		resp, err := getAndWaitResponse(rt, req, 10*time.Second, http.StatusForbidden)
+		framework.ExpectNoError(err)
+		Expect(resp.StatusCode).To(Equal(http.StatusForbidden))
+
+		By(fmt.Sprintf("Calling ingress %s/%s we wait to get a 200 with opaServeResponse %s policy", ingressUpdate.Namespace, ingressUpdate.Name, opaPolicyName))
+		req.Header.Set("Authorization", "Basic permissions_token") // Authorized request
+		resp, err = getAndWaitResponse(rt, req, 10*time.Second, http.StatusOK)
+		framework.ExpectNoError(err)
+		Expect(resp.StatusCode).To(Equal(http.StatusOK))
+		s, err := getBody(resp)
+		framework.ExpectNoError(err)
+		Expect(s).To(ContainSubstring(expectedPermissionsContent))
+	})
+
+})
+
+func createIngressWithInfo(serviceName, hostName, ns string, port int, cs kubernetes.Interface, jig *ingress.TestJig) *netv1.Ingress {
+	labels := map[string]string{"app": serviceName}
+	waitTime := 10 * time.Minute
+
+	// Create initial ingress
+	ing := createIngress(serviceName, hostName, ns, "/", netv1.PathTypeImplementationSpecific, labels, nil, port)
+	var err error
+	ingressCreate, err := cs.NetworkingV1().Ingresses(ns).Create(context.TODO(), ing, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	// Wait for ingress address
+	addr, err := jig.WaitForIngressAddress(context.TODO(), cs, ns, ingressCreate.Name, waitTime)
+	framework.ExpectNoError(err)
+
+	// Ensure ingress exists
+	_, err = cs.NetworkingV1().Ingresses(ns).Get(context.TODO(), ing.Name, metav1.GetOptions{ResourceVersion: "0"})
+	framework.ExpectNoError(err)
+
+	// Verify skipper routing
+	By("Waiting for skipper route to default redirect from http to https")
+	err = waitForResponse(addr, "http", waitTime, isRedirect, true)
+	framework.ExpectNoError(err)
+
+	By("Waiting for ALB to create endpoint and skipper route")
```

Comment:
This is not ALB. Maybe change to more generic message like: `Waiting for LB and skipper routes to be ready`